### DISCUSSION TITLE
about sealed and growing segments for small segment

### DISCUSSION THREAD

**Discussion by wookiist** [milvus-io/milvus#31075] [Q&A and General discussion] [answered]
DATA COORDINATOR performs a SEAL operation on GROWING SEGMENTS that are created above a certain threshold in size. 
So, if there is a newly created collection and data below the sealing threshold is inserted into this collection, the growing segment of this collection cannot be converted to a sealed segment forever?

**Comment by wookiist** [comment]
self answering.

https://medium.com/vector-database/data-insertion-and-data-persistence-in-a-vector-database-a27894d6962e
> Growing segments that are not sealed after a long period of time will be sealed as too many growing segments cause data nodes to over-consume memory.

**Reply by wookiist** [reply]
But what is the factor of "a long period of time"?

**Comment by yhmo** [comment, marked as answer]
If a segment didn't accept dml records in maxIdleTime and the size of segment is greater than minSizeFromIdleToSealed, Milvus will automatically seal it. The default is 600 seconds.
https://github.com/milvus-io/milvus/blob/d0eeea4b443f5353421de69a22d83dc1fa4dee23/configs/milvus.yaml#L411

**Reply by xiaofan-luan** [reply]
is that automatically seal? 
I thought 600 s is the automatic sync time. and seal time could be up to one day

**Reply by yhmo** [reply]
Yes, "sync" is another concept to explain..... 
From the source code, 4 policies are defined for "seal", including maxIdleTime and maxLife: https://github.com/milvus-io/milvus/blob/2f4a13a7ae84d1d12994593e7fb09f680d33b658/internal/datacoord/segment_manager.go#L188
https://github.com/milvus-io/milvus/blob/2f4a13a7ae84d1d12994593e7fb09f680d33b658/internal/datacoord/segment_manager.go#L622

In the source code, looks like "seal" is "sync".
Then the description of maxIdleTime is not proper:
```
    maxLife: 86400 # The max lifetime of segment in seconds, 24*60*60
    # If a segment didn't accept dml records in maxIdleTime and the size of segment is greater than
    # minSizeFromIdleToSealed, Milvus will automatically seal it.
    # The max idle time of segment in seconds, 10*60.
    maxIdleTime: 600
    minSizeFromIdleToSealed: 16 # The min size in MB of segment which can be idle from sealed.
```
